runtime/Cpp/runtime/src/misc/IntervalSet.h

```diff
@@ -30,15 +31,17 @@ namespace misc {
   protected:
     /// The list of sorted, disjoint intervals.
     std::vector<Interval> _intervals;
-    bool _readonly;
+    std::atomic<bool> _readonly;
```

Comment:
I comment on this in the bug report. The atomic<bool> alone is not sufficient to remove the race condition around updates. This is a pattern we use a lot.